File: app/src/main/java/com/gandhasiri/app/TreeViewModel.java
package com.gandhasiri.app;

import android.app.Application;
import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import java.util.List;

public class TreeViewModel extends AndroidViewModel {
    private TreeRepository repository;
    private LiveData<List<Tree>> allTrees;

    public TreeViewModel(@NonNull Application application) {
        super(application);
        repository = new TreeRepository(application);
        allTrees = repository.getAllTrees();
    }

    public LiveData<List<Tree>> getAllTrees() {
        return allTrees;
    }

    public LiveData<Tree> getTreeById(int id) {
        return repository.getTreeById(id);
    }

    public void insert(Tree tree, double initialGirth) {
        repository.insertWithLog(tree, initialGirth);
    }

    public void updateTreeGirth(Tree tree, double newGirth) {
        repository.updateTreeWithLog(tree, newGirth);
    }

    public void delete(Tree tree) {
        repository.delete(tree);
    }

    public LiveData<List<GrowthLog>> getGrowthLogs(int treeId) {
        return repository.getGrowthLogs(treeId);
    }
}

File: app/src/main/java/com/gandhasiri/app/TreeViewModel.kt
package com.gandhasiri.app

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

class TreeViewModel(application: Application) : AndroidViewModel(application) {
    private val repository: TreeRepository = TreeRepository(application)
    val allTrees: LiveData<List<Tree>> = repository.getAllTrees()

    fun getTreeById(id: Int): LiveData<Tree> = repository.getTreeById(id)

    fun insert(tree: Tree, initialGirth: Double) {
        repository.insertWithLog(tree, initialGirth)
    }

    fun updateTreeGirth(tree: Tree, newGirth: Double) {
        repository.updateTreeWithLog(tree, newGirth)
    }

    fun delete(tree: Tree) {
        repository.delete(tree)
    }

    fun getGrowthLogs(treeId: Int): LiveData<List<GrowthLog>> = repository.getGrowthLogs(treeId)
}
